packages/doc/docs/quickstart.mdx

```diff
@@ -0,0 +1,444 @@
+
+    ```sql
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = 'LIUKA'
+    LIMIT 1
+    ```
+
+    We'd like to use the name of the user's input `name` to replace the static string “LIUKA”, so please update the SQL using the [template variable](./api-building/sql-template#dynamic-parameter) `{{ context.params.name }}`.
+
+    ```sql
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name }}
+    LIMIT 1
+    ```
+
+    Your first API with parameters input is done! Try the API by visiting it:
+
+    - [localhost:3000/api/customer?name=GABEY](http://localhost:3000/api/customer?name=GABEY)
+    - [localhost:3000/api/customer?name=KENDELL](http://localhost:3000/api/customer?name=KENDELL)
+
+    <br />
+
+    :::tip
+    Is it safe to render data from external sources like the user's input?
+
+    Yes, we'll parameterize all the user input like the below to prevent SQL injections.
+    ![screenshot of parameterd queries](./img/parametered-queries.png)
+
+    Check [Display the variable](./api-building/sql-template#display-the-variable--dynamic-parameter) for more information.
+    :::
+
+2.  Apply a [filter](./api-building/sql-template#filters) to the input, we can to `upper` filter to let our input be case-insensitive.
+
+    ```sql
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    LIMIT 1
+    ```
+
+    Now, test with some names in lower case:
+
+    - [http://localhost:3000/api/customer?name=kendell](http://localhost:3000/api/customer?name=kendell)
+
+3.  Throw error when user not found. We can let our API better by showing an accurate error message.
+    First, we need to execute an extra query to know whether the user is in our database:
+
+    ```sql
+    {% req user %}
+    SELECT COUNT(\*) AS count FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    {% endreq %}
+
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    LIMIT 1
+    ```
+
+    The block `{% req user %} ... {% endreq %}` is a [query block](./api-building/sql-builder) which tells VulcanSQL that we want to save the query result to `user` variable instead of outputting as responses.
+
+    Now we can get the result of this query and throw an error when the result equals `0`.
+
+    ```sql
+    {% req user %}
+    SELECT COUNT(*) AS count FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    {% endreq %}
+
+    {% if user.value()[0].count == 0 %}
+      {% error "CUSTOMER_NOT_FOUND" %}
+    {% endif %}
+
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    LIMIT 1
+    ```
+
+    We used the if expression to throw an error when the result equals `0`, VulcanSQL server will stop executing and respond immediately when meeting a {% error %} tag, `"CUSTOMER_NOT_FOUND"` is the error code we want to throw.
```

Comment:
Can we add the quote to be `{% error %}` ?

Reply:
Sure